WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        let blog = apost.blog
+        
+        if blog.supports(.Stats) {
+            // Needs Jetpack.
+            return
+        }
+        
+        WPAnalytics.track(.PostListStatsAction, withProperties: propertiesForAnalytics())
+        
+        // Push the Stats Post Details ViewController
+        let identifier = NSStringFromClass(StatsPostDetailsTableViewController.self)
+        let service = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let statsBundle = NSBundle(forClass: WPStatsViewController.self)
+        let path = statsBundle.pathForResource("WordPressCom-Stats-iOS", ofType: "bundle")!
+        let bundle = NSBundle(path: path)
+        let statsStoryboard = UIStoryboard(name: self.dynamicType.statsStoryboardName, bundle: bundle)
+        let controller = statsStoryboard.instantiateViewControllerWithIdentifier(identifier) as! StatsPostDetailsTableViewController
+        
+        controller.postID = apost.postID
+        controller.postTitle = apost.titleForDisplay()
+        controller.statsService = WPStatsService(siteId: blog.dotComID, siteTimeZone: service.timeZoneForBlog(blog), oauth2Token: blog.authToken, andCacheExpirationInterval: self.dynamicType.statsCacheInterval)
+        
+        navigationController?.pushViewController(controller, animated: true)
+    }
+    
+    // MARK: - Filter Related
+    
+    override internal func currentPostAuthorFilter() -> PostAuthorFilter {
+        if !canFilterByAuthor() {
+            // No REST API, so we have to use XMLRPC and can't filter results by author.
+            return .Everyone
+        }
+        
+        if let filter = NSUserDefaults.standardUserDefaults().objectForKey(self.dynamicType.currentPostAuthorFilterKey) {
+            if filter.unsignedIntegerValue == PostAuthorFilter.Everyone.rawValue {
+                return .Everyone
+            }
+        }
+        
+        return .Mine
+    }
+
+    override internal func setCurrentPostAuthorFilter(filter: PostAuthorFilter) {
+        guard filter != currentPostAuthorFilter() else {
+            return
+        }
+        
+        WPAnalytics.track(.PostListAuthorFilterChanged, withProperties: propertiesForAnalytics())
+        
+        NSUserDefaults.standardUserDefaults().setObject(filter.rawValue, forKey: self.dynamicType.currentPostAuthorFilterKey)
+        NSUserDefaults.resetStandardUserDefaults()
+        
+        recentlyTrashedPostObjectIDs?.removeAllObjects()
+        resetTableViewContentOffset()
+        updateAndPerformFetchRequestRefreshingCachedRowHeights()
+        syncItemsWithUserInteraction(false)
+    }
+
+    private func keyForCurrentListStatusFilter() -> String {
+        return self.dynamicType.currentPostListStatusFilterKey
+    }
+
+    // MARK: - Cell Delegate Methods
+
+    func cell(cell: UITableViewCell!, receivedEditActionForProvider contentProvider: WPPostContentViewProvider!) {
+        let apost = contentProvider as! AbstractPost
```

Comment:
> I know that contentProvider should only ever be an AbstactPost but for paranoia it would be nice to avoid the explicitly unwrapped optional. 

I'd rather not have things that are clear mistakes on our end failing silently.  I think that's a recipe for not noticing logic errors.

The real issue in these lines of code is that we shouldn't need to cast `WPPostContentViewProvider` to `AbstractPost`.  It means something's not right - but fixing that is outside the scope of this PR (ie: convert 1:1).

> It would be nice to encapsulate the delete methods into an extension.

Will do.

> I also think its reasonable for this PR to edit PostCardTableViewCellDelegate.h and add `NS_ASSUME_NONNULL_BEGIN` `NS_ASSUME_NONNULL_END`.

Will do.